torchgeo/datasets/dota.py

```diff
@@ -0,0 +1,505 @@
+                },
+                '1.5': {
+                    'filename': 'dotav1.0_images_train.tar.gz',
+                    'md5': '363b472dc3c71e7fa2f4a60223b437ea',
+                },
+                '2.0': {
+                    'filename': 'dotav2.0_images_train.tar.gz',
+                    'md5': '91ae5212d170330ab9f65ccb6c675763',
+                },
+            },
+            'annotations': {
+                '1.0': {
+                    'filename': 'dotav1.0_annotations_train.tar.gz',
+                    'md5': 'f6788257bcc4d29018344a4128e3734a',
+                },
+                '1.5': {
+                    'filename': 'dotav1.5_annotations_train.tar.gz',
+                    'md5': '0da97e5623a87d7bec22e75f6978dbce',
+                },
+                '2.0': {
+                    'filename': 'dotav2.0_annotations_train.tar.gz',
+                    'md5': '04d3d626df2203053b7f06581b3b0667',
+                },
+            },
+        },
+        'val': {
+            'images': {
+                '1.0': {
+                    'filename': 'dotav1.0_images_val.tar.gz',
+                    'md5': '42293219ba61d61c417ae558bbe1f2ba',
+                },
+                '1.5': {
+                    'filename': 'dotav1.0_images_val.tar.gz',
+                    'md5': '42293219ba61d61c417ae558bbe1f2ba',
+                },
+                '2.0': {
+                    'filename': 'dotav2.0_images_val.tar.gz',
+                    'md5': '737f65edf54b5aa627b3d48b0e253095',
+                },
+            },
+            'annotations': {
+                '1.0': {
+                    'filename': 'dotav1.0_annotations_val.tar.gz',
+                    'md5': '28155c05b1dc3a0f5cb6b9bdfef85a13',
+                },
+                '1.5': {
+                    'filename': 'dotav1.5_annotations_val.tar.gz',
+                    'md5': '85bf945788784cf9b4f1c714453178fc',
+                },
+                '2.0': {
+                    'filename': 'dotav2.0_annotations_val.tar.gz',
+                    'md5': 'ec53c1dbcfc125d7532bd6a065c647ac',
+                },
+            },
+        },
+    }
+
+    sample_df_path = 'samples.csv'
+
+    classes = (
+        'plane',
+        'ship',
+        'storage-tank',
+        'baseball-diamond',
+        'tennis-court',
+        'basketball-court',
+        'ground-track-field',
+        'harbor',
+        'bridge',
+        'large-vehicle',
+        'small-vehicle',
+        'helicopter',
+        'roundabout',
+        'soccer-ball-field',
+        'swimming-pool',
+        'container-crane',
+        'airport',
+        'helipad',
+    )
+
+    valid_splits = ('train', 'val')
+    valid_versions = ('1.0', '1.5', '2.0')
+
+    valid_orientations = ('horizontal', 'oriented')
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        version: str = '2.0',
+        bbox_orientation: str = 'oriented',
```

Comment:
```suggestion
        split: Literal['train', 'val'] = 'train',
        version: Literal['1.0', '1.5', '2.0'] = '2.0',
        bbox_orientation: Literal['horizontal', 'oriented'] = 'oriented',
```
Using `typing.Literal` to properly document allowed inputs. Not all strings are valid inputs.